### DISCUSSION TITLE
Does Post Type Switcher store the redirection information?

### DISCUSSION THREAD

**Discussion by WindingWinter** [johnjamesjacoby/post-type-switcher#41] [Q&A] [answered]
I use Post Type Switcher to switch some of my blog post type from 'post' to 'docs' ( a custom post type), thus the URL is changed from `/blog/* `to` /docs/*`. I note that the original post at /blog/*( eg: `/blog/someinfo` is automatically (301) redirect to` /docs/* `( eg: from `/blog/someinfo` to `/docs/someinfo`). When I use a redirection check website, I found that the redirection is done on WordPress level. 

Changing the post type from 'post' to 'page' also has the same effect.

I thought that this is well explained by [the WordPress creates redirect from old URL after changing slug question](https://wordpress.stackexchange.com/q/44411). So I use [SELECT * FROM wp_postmeta where meta_key = "_wp_old_slug"](https://wordpress.stackexchange.com/questions/44411/when-i-change-slug-will-wordpress-create-redirect-from-the-old-url#comment574496_44481) to confirm that the redirection is indeed stored there. But to my huge surprise, I can't find anything that has anything to do with the above redirection.

( If on the other hand, I just change the slug manually and not changing the Post Type, like changing the url from `/blog/someinfo` to `/blog/someinfo-2`, then I would see something at meta_value column, and hence I can rest assure that the redirection is real and permanent).

This creates a confusion for me. Does Post Type Switcher store the redirection information in the WordPress database? If yes, at where? I go through [the source code](https://github.com/JJJ/post-type-switcher/blob/master/post-type-switcher.php), and it seems that the plugin only changes the database only once, at one type, at [set_post_type](https://github.com/JJJ/post-type-switcher/blob/master/post-type-switcher.php#L428), so no redirection information is written.

How the redirection is accomplished? Anyone can shed any lights on this?

**Comment by JJJ** [comment, marked as answer]
Great question @WindingWinter!

Here's how WordPress handles this (without the Post Type Switcher needing to do anything)

https://developer.wordpress.org/reference/functions/redirect_canonical/

It's pretty awesome!
